crates/iceberg-catalog/src/service/storage/mod.rs

```diff
@@ -34,6 +34,9 @@ use crate::{
     Debug, Clone, Eq, PartialEq, Serialize, Deserialize, derive_more::From, utoipa::ToSchema,
 )]
 #[serde(tag = "type", rename_all = "kebab-case")]
+#[allow(clippy::unsafe_derive_deserialize)]
+// tokio::join! uses unsafe code internally.
+// This is no problem for us as we don't circumvent the initialization of the struct.
```

Comment:
```suggestion
// tokio::join! uses unsafe code internally.
// This is no problem since our constructor does not enforce any invariants relevant to the unsafe code. Deserialize is even the primary way of constructing `StorageProfile` since it is received via REST.
```